charts/aws-vpc-cni/templates/daemonset.yaml

```diff
@@ -183,6 +183,7 @@ spec:
       - name: xtables-lock
         hostPath:
           path: /run/xtables.lock
+          type: FileOrCreate
```

Comment:
Should we have reasons to believe that `/run/xtables.lock` *file* should be have been created by the iptables package in the operating system?

There is note of caution in kubelet volume docs - https://kubernetes.io/docs/concepts/storage/volumes/

> Caution: The FileOrCreate mode does not create the parent directory of the file. If the parent directory of the mounted file does not exist, the pod fails to start. To ensure that this mode works, you can try to mount directories and files separately, as shown in the [FileOrCreate example](https://kubernetes.io/docs/concepts/storage/volumes/#hostpath-fileorcreate-example) for hostPath.

Reply:
Just tested this on a fresh Ubuntu 22.04 VM. This file is not created until the first run of the `iptables-legacy` command.

The default `iptables` is a compatibility wrapper around nftables, which never creates this file at all, it is only created when interacting with actual iptables with `iptables-legacy`. So if nothing on startup calls the legacy iptables interface, the file will not exist.

Reply:
> If the parent directory of the mounted file does not exist, the pod fails to start. To ensure that this mode works, you can try to mount directories and files separately, as shown in the [FileOrCreate example](https://kubernetes.io/docs/concepts/storage/volumes/#hostpath-fileorcreate-example) for hostPath.

I would expect `/run` to exist on all Linux hosts, as it is part of [FHS](https://refspecs.linuxfoundation.org/FHS_3.0/fhs/ch03s15.html), so I don't think this is a risk? `kube-proxy` (as set up by `kubeadm`) already uses `FileOrCreate` for this path, so if `/run` doesn't exist, `kube-proxy` won't start anyway, and the node won't function properly.